tools/clang/tools/dxclib/dxc.cpp

```diff
@@ -1216,6 +1218,36 @@ void DxcContext::GetCompilerVersionInfo(llvm::raw_string_ostream &OS) {
 #define VERSION_STRING_SUFFIX ""
 #endif
 
+#ifdef _WIN32
+// Unhandled exception filter called when an unhandled exception occurs
+// to at least print an generic error message instead of crashing silently.
+// passes exception along to allow crash dumps to be generated
+static LONG CALLBACK ExceptionFilter(PEXCEPTION_POINTERS pExceptionInfo)
+{
+  if (pExceptionInfo && pExceptionInfo->ExceptionRecord) {
+    switch(pExceptionInfo->ExceptionRecord->ExceptionCode) {
+    case EXCEPTION_ACCESS_VIOLATION: {
+      const char *readWrite;
+      if (pExceptionInfo->ExceptionRecord->ExceptionInformation[0])
+        readWrite = "write";
+      else
+        readWrite = "read";
+      std::cerr <<
```

Comment:
Let's use `fputs` here to avoid dragging in c++ streams here. Might not be in a great state in the exception handler.